src/sessions.ts

```diff
@@ -823,16 +856,22 @@ export class ClientSession
                     backoffMS > this.timeoutContext.remainingTimeMS) ||
                   now() + backoffMS > startTime + MAX_TIMEOUT;
 
+                // 6.ii (cont.) If elapsed time + backoffMS > TIMEOUT_MS, then raise last known error
+                // DP: the lines above do this math slightly indirectly, but this break does NOT raise the last known error
+                // because breaking the while loop takes us to the return result statement outside the while(!committed)
```

Comment:
actually, to the top of the for retry loop (but with the same effect re: last known error)